### PR TITLE
feat: Add Switch Page

### PR DESCRIPTION
**Pull request by williamernest**
fixes: #31

### CODE REVIEW

File: config/webpack.config.prod.js [hunk 1]
```diff
@@ -47,7 +47,7 @@ const extractTextPluginOptions = shouldUseRelativeAssetPaths
   : {};
 
 // List of all component names for generating the corresponding html pages.
-const components = ['button'];
+const components = ['button', 'switch'];
```
Comment (moog16) [inline config/webpack.config.prod.js]: We put this on a separate line

Comment (moog16) [reply config/webpack.config.prod.js]: Also add switch to config.dev.js file

File: package.json [hunk 2]
```diff
@@ -105,5 +105,8 @@
   },
   "eslintConfig": {
     "extends": "react-app"
+  },
+  "dependencies": {
+    "@material/switch": "^0.35.0"
```
Comment (moog16) [inline package.json]: uninstall and install as devDep

File: src/SwitchPage.js [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+        designLink='https://material.io/guidelines/components/buttons.html'
+        docsLink='https://material.io/components/web/catalog/buttons/'
+        sourceLink='https://github.com/material-components/material-components-web/tree/master/packages/mdc-button'
+        demos={<SwitchDemos/>}
+      />
+    </div>
+  );
+}
+
+class SwitchHero extends Component {
```
Comment (moog16) [inline src/SwitchPage.js]: Component should be functional if no lifecycle methods

File: src/images/ic_switch_24px.svg [hunk 4]
```diff
@@ -0,0 +1 @@
...95 9.45 27.63 22.57 32.11 1.65.31 2.25-.74 2.25-1.63 0-.8-.03-2.93-.04-5.75-9.18 2.04-11.12-4.54-11.12-4.54-1.5-3.91-3.66-4.95-3.66-4.95-3-2.1.23-2.06.23-2.06 3.31.24 5.05 3.49 5.05 3.49 2.94 5.17 7.72 3.68 9.6 2.81.3-2.19 1.15-3.68 2.09-4.52-7.33-.85-15.03-3.76-15.03-16.72 0-3.69 1.29-6.71 3.4-9.08-.34-.86-1.47-4.3.32-8.95 0 0 2.77-.91 9.08 3.47 2.63-.75 5.46-1.13 8.26-1.14 2.8.01 5.63.39 8.26 1.14 6.3-4.38 9.07-3.47 9.07-3.47 1.8 4.66.67 8.1.33 8.95 2.11 2.37 3.39 5.39 3.39 9.08 0 13-7.72 15.86-15.07 16.7 1.19 1.04 2.24 3.11 2.24 6.27 0 4.52-.04 8.17-.04 9.28 0 .91.59 1.96 2.27 1.63 13.1-4.48 22.55-17.16 22.55-32.1.01-18.71-14.77-33.86-33-33.86"/></defs><clipPath id="b"><use xlink:href="#a" overflow="visible"/></clipPath><g clip-path="url(#b)"><defs><path id="c" d="M-1121.33-3307.17h1680v3369h-1680z"/></defs><clipPath id="d"><use xlink:href="#c" overflow="visible"/></clipPath><path clip-path="url(#d)" d="M-278.33-858.17h76v76h-76z"/></g><path d="M19 9h-8.02C10.06 7.79 8.63 7 7 7c-2.76 0-5 2.24-5 5s2.24 5 5 5c1.63 0 3.06-.79 3.98-2H19c1.66 0 3-1.34 3-3s-1.34-3-3-3zm0 4h-7.1c.07-.32.1-.66.1-1s-.04-.68-.1-1H19c.55 0 1 .45 1 1s-.45 1-1 1z"/><path fill="none" d="M0 0h24v24H0z"/></svg>
```
Comment (moog16) [inline src/images/ic_switch_24px.svg]: use the https://drive.google.com/corp/drive/folders/19cJSLNDYJ2TF1Rv-Ch2KpY9eizy_EHGF images

File: src/SwitchPage.js [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+const SwitchPage = () => {
+  return (
+    <div>
+      <HeaderBar />
+      <ComponentPage
+        hero={<SwitchHero/>}
+        title='Switch'
+        description='Switchs communicate an action a user can take. They are typically placed throughout your UI, in places like dialogs, forms, cards, and toolbars.'
+        designLink='https://material.io/guidelines/components/buttons.html'
+        docsLink='https://material.io/components/web/catalog/buttons/'
```
Comment (moog16) [inline src/SwitchPage.js]: https://material.io/components/web/catalog/input-controls/switches/

File: src/SwitchPage.js [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+
+const SwitchPage = () => {
+  return (
+    <div>
+      <HeaderBar />
+      <ComponentPage
+        hero={<SwitchHero/>}
+        title='Switch'
+        description='Switchs communicate an action a user can take. They are typically placed throughout your UI, in places like dialogs, forms, cards, and toolbars.'
+        designLink='https://material.io/guidelines/components/buttons.html'
```
Comment (moog16) [inline src/SwitchPage.js]: this one seems wrong

File: src/SwitchPage.js [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+        docsLink='https://material.io/components/web/catalog/buttons/'
+        sourceLink='https://github.com/material-components/material-components-web/tree/master/packages/mdc-button'
```
Comment (moog16) [inline src/SwitchPage.js]: https://github.com/material-components/material-components-web/tree/master/packages/mdc-switch

File: src/SwitchPage.js [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+      />
+    </div>
+  );
+}
+
+class SwitchHero extends Component {
+  render() {
+    return (
+        <div>
+          <div className='mdc-switch'>
```
Comment (moog16) [inline src/SwitchPage.js]: This one is not working on the page

File: src/SwitchPage.js [hunk 9]
```diff
@@ -0,0 +1,80 @@
...
+          <label htmlFor={`${title}-switch`}>off/on</label>
+        </div>
+    );
+  }
+
+  render() {
+    return (
+      <div>
+        {this.renderSwitchVariant('Switch')}
+        {this.renderSwitchVariant('Disabled Switch', 'disabled')}
```
Comment (lynnjepsen) [inline src/SwitchPage.js]: drop the disabled switch

File: src/styles/SwitchPage.scss [hunk 10]
```diff
@@ -0,0 +1 @@
+@import "@material/switch/dist/mdc.switch";
```
Comment (lynnjepsen) [inline src/styles/SwitchPage.scss]: You need more padding between the "on/off" and the switch

**Pull request merged**